cats-effect/src/test/scala/neotypes/cats/effect/IOSuite.scala

```diff
@@ -0,0 +1,21 @@
+package neotypes.cats.effect
+
+import cats.effect.IO
+import neotypes.{Async, EffectSuite, EffectTestkit}
+import neotypes.cats.effect.implicits._
+import scala.concurrent.{ExecutionContext, Future}
+
+/** Implementation of the Effect Teskit for cats-effect IO. */
+object IOTestkit extends EffectTestkit[IO] {
+  override def createBehaviour(implicit ec: ExecutionContext): Behaviour =
+    new Behaviour {
+      override final def fToFuture[T](io: IO[T]): Future[T] =
+        io.unsafeToFuture()
+
+      override final val asyncInstance: Async[IO] =
+        implicitly
+    }
+}
+
+/** Execute all the effect specs using IO. */
+final class IOSuite extends EffectSuite(IOTestkit)
```

Comment:
When I run a test suite the output in sbt for the suite looks like this:
```java
[info] EffectSuite$$anon$6:

[info] EffectSuite$$anon$7:

[info] EffectSuite$$anon$8:
```
It would be a bit nicer to see AsyncGuaranteeSpec, AsyncIntegrationSpec, etc...

Reply:
Yeah, that is annoying but after that line, you can see the behavior line.
For example, [here](https://travis-ci.org/github/neotypes/neotypes/builds/679884593#L1131):

> [info] EffectSuite$$anon$1:
> [info] Async[scala.concurrent.Future].guarantee

That anonymous class name comes from [here](https://github.com/neotypes/neotypes/pull/119/files#diff-bc4e76c29a5d67a8f6b773b289d50ea5R36-R47):

```scala
abstract class EffectSuite[F[_]](testkit: EffectTestkit[F]) extends Suites(
  new AsyncGuaranteeSpec(testkit) {},
  new AsyncIntegrationSpec(testkit) {},
  new BasicSessionSpec(testkit) {},
  new BasicTransactionSpec(testkit) {},
  new CompositeTypesSpec(testkit) {},
  new ParameterSpec(testkit) {},
  new PathSessionSpec(testkit) {},
  new QueryExecutionSpec(testkit) {},
  new StageSyntaxSpec(testkit) {},
  new TransactIntegrationSpec(testkit) {}
)
```

The problem is that if I use concrete classes, then **Scalatests** tries to run them.

AFAIK, there is no way to solve this.
But will check once again.

Reply:
@geoffjohn11 oh it seems that since now the specs have a constructor, **Scalatest** doesn't try to run it. So now I am replacing the abstract classes with final classes and the names are good again :smile: